sklearnex/covariance/incremental_covariance.py

```diff
@@ -14,76 +14,157 @@
 
     Parameters
     ----------
+    store_precision : bool, default=False
+        Specifies if the estimated precision is stored.
+
+    assume_centered : bool, default=False
+        If True, data are not centered before computation.
+        Its use is highly discouraged and will reduce performance.
+        If False (default), data are centered before computation.
+
     batch_size : int, default=None
         The number of samples to use for each batch. Only used when calling
         ``fit``. If ``batch_size`` is ``None``, then ``batch_size``
         is inferred from the data and set to ``5 * n_features``, to provide a
         balance between approximation accuracy and memory consumption.
 
+    copy : bool, default=True
+        If False, X will be overwritten. ``copy=False`` can be used to
+        save memory but is unsafe for general use.
+
     Attributes
     ----------
     location_ : ndarray of shape (n_features,)
         Estimated location, i.e. the estimated mean.
 
     covariance_ : ndarray of shape (n_features, n_features)
         Estimated covariance matrix
+
+    n_samples_seen_ : int
+        The number of samples processed by the estimator. Will be reset on
+        new calls to fit, but increments across ``partial_fit`` calls.
+
+    batch_size_ : int
+        Inferred batch size from ``batch_size``.
+
+    n_features_in_ : int
+        Number of features seen during :term:`fit` `partial_fit`.
     """
 
     _onedal_incremental_covariance = staticmethod(onedal_IncrementalEmpiricalCovariance)
 
-    def __init__(self, batch_size=None):
-        self._need_to_finalize = False  # If True then finalize compute should
-        #      be called to obtain covariance_ or location_ from partial compute data
+    if sklearn_check_version("1.2"):
+        _parameter_constraints: dict = {
+            "store_precision": ["boolean"],
+            "assume_centered": ["boolean"],
+            "batch_size": [Interval(numbers.Integral, 1, None, closed="left"), None],
+            "copy": ["boolean"],
+        }
+
+    get_precision = sklearn_EmpiricalCovariance.get_precision
+    error_norm = wrap_output_data(sklearn_EmpiricalCovariance.error_norm)
+    score = wrap_output_data(sklearn_EmpiricalCovariance.score)
+
+    def __init__(
+        self, *, store_precision=False, assume_centered=False, batch_size=None, copy=True
+    ):
+        self.assume_centered = assume_centered
+        self.store_precision = store_precision
         self.batch_size = batch_size
+        self.copy = copy
+
+    def _onedal_supported(self, method_name, *data):
+        patching_status = PatchingConditionsChain(
+            f"sklearn.covariance.{self.__class__.__name__}.{method_name}"
+        )
+        return patching_status
 
     def _onedal_finalize_fit(self):
         assert hasattr(self, "_onedal_estimator")
         self._onedal_estimator.finalize_fit()
         self._need_to_finalize = False
 
+        if self.assume_centered:
```

Comment:
I don't think it makes sense in this way. in sklearn.EmpiricalCovariance this flag is used to simplify computations when data are assumed to be centered. but here it causes additional computations. IMO it is better to remove it until we will add this feature on oneDAL side.

Reply:
There exists a logic issue here then. Either it is like IncrementalPCA, where all the methods of PCA are supported (in this case for EmpiricalCovariance) or the class should be named differently.  I believe this class should have the same inputs as EmpiricalCovariance. If a user tries to use this class thinking it can be a drop-in replacement for EmpiricalCovariance, they will face issues. There is a warning in the class documentation not to use it, but is available as necessary to the user. It is mathematically valid, just a poor choice to use it.

Reply:
i feel that it is not good choice to add something which shouldn't be used. then it is better to do nothing and just add it to args for API conformance (with highlighting that fact in the docstring). have to notice that if data is really centered then output for both cases is the same, so it is weird to do redundant computations. i highly doubt that someone uses covariance estimator to compute scalar product of vectors. more likely, the user wants to compute covariance faster setting assume_centered=True and doing nothing special is more preferrable in this case

Reply:
also have to add that IMO the API for this class shouldn't be exactly the same as for EmpiricalCovariance, it is just assumption which can be ignored at this point if it creates some problems.

Reply:
ok, probably removing the parameter will cause issues with passing tests for sklearn.EmpiricalCovariance, so, doing nothing must be the best way

Reply:
https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/covariance/_empirical_covariance.py#L107  its a rather simple thing that I would expect wouldn't be worth the work (and I am not sure why they implemented it in the first place since its rather trivial). I guess I am just hesitant about missing out on additional tests because it no longer conforms to sklearn's empirical covariance.  As a compromise, I'd suggest we add a warning just before the math is applied.

let me know what you think

Reply:
> i feel that it is not good choice to add something which shouldn't be used. then it is better to do nothing and just add it to args for API conformance (with highlighting that fact in the docstring). have to notice that if data is really centered then output for both cases is the same, so it is weird to do redundant computations. i highly doubt that someone uses covariance estimator to compute scalar product of vectors. more likely, the user wants to compute covariance faster setting assume_centered=True and doing nothing special is more preferrable in this case

this argument is still valid for me. i think that the best way is to just ignore this parameter with marking it as added for API conformance, otherwise, it will force the user to remove assume_centered=True from estimator calls in their external code to reach the best performance (that's probably exactly what they wanted setting this parameter)

Reply:
I would like to say that its not extremely hard to add this support on cpp(oneDAL) side firstly, instead of using additional computations, due to user will face with degradation, when speedup is expected